test/global-setup.spec.ts

```diff
@@ -88,7 +88,30 @@ test('globalTeardown does not run when globalSetup times out', async ({ runInlin
       });
     `,
   });
-  expect(result.skipped).toBe(1);
+  // We did not collect tests, so everything should be zero.
+  expect(result.skipped).toBe(0);
+  expect(result.passed).toBe(0);
+  expect(result.failed).toBe(0);
   expect(result.output).toContain('Timed out waiting 1s for the entire test run');
   expect(result.output).not.toContain('teardown=');
 });
+
+test('globalSetup should be run before requiring tests', async ({ runInlineTest }) => {
+  const { passed } = await runInlineTest({
+    'folio.config.ts': `
+      folio.globalSetup(async () => {
+        process.env.FOO = JSON.stringify({ foo: 'bar' });
```

Comment:
I am still a bit confused by the setup thought. Tests require the config, but the config runs before the tests are required?? And global setup is async, so who awaits it? You set process.env. Is that because the global setup is run out of process?

If I did `global.foo = 123` in the globalSetup, could I read it in the test file?

Reply:
- Global setup is executed in the main process - setting globals will not be visible in workers.
- Config runs at any moment (whenever someone requires it), but `folio.globalSetup` registers a function that is run at some predefined moment by the test runner.